models/repo_list.go

```diff
@@ -228,7 +228,7 @@ func SearchRepository(opts *SearchRepoOptions) (RepositoryList, int64, error) {
 		}
 
 		if opts.AllPublic {
-			accessCond = accessCond.Or(builder.Eq{"is_private": false})
+			accessCond = accessCond.Or(builder.Eq{"is_private": false}).And(builder.In("owner_id", builder.Select("id").From("user").Where(builder.Lte{"visibility": 1})))
```

Comment:
Does this work for user repositories?

Reply:
Suer, default value is 0 and user dont use this flag so user are always public

Reply:
And if we implement visabilety for users, it will handle out of the box ;)